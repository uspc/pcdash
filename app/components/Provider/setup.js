import storage from '../../libs/storage';
import persist from '../../libs/persist';
import NoteStore from '../../stores/NoteStore';
import LaneStore from '../../stores/LaneStore';
import loadseed from '../../libs/loadseed';

export default alt => {
  alt.addStore('NoteStore', NoteStore);
  alt.addStore('LaneStore', LaneStore);

  loadseed(sessionStorage,'app',alt);
  //persist(alt, storage(sessionStorage), 'app');
}


