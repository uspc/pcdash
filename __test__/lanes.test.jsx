import renderer from 'react-test-renderer';
import React from 'react';
import Lanes from '../app/components/Lanes.jsx';
import Provider from '../app/components/Provider'
import {DragDropContext} from 'react-dnd';
import TestBackend from 'react-dnd-test-backend';

        test('Testing Lanes - Initial', () => {
                var lanesmap = [{id:1,editing:true,name:"testing"}];
                const LanesContent = wrapTestContext(Lanes);
                const tree = renderer.create(<Provider><LanesContent lanes={lanesmap}/></Provider>).toJSON();
                expect(tree).toMatchSnapshot();
        });


function wrapTestContext(DecoratedComponent){
        return DragDropContext (TestBackend)(
            React.createClass({
                    render(){
                            return <DecoratedComponent {...this.props} />;
                    }
            })
        );
}