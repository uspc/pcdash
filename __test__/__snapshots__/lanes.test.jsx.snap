exports[`test Testing Lanes - Initial 1`] = `
<div
  className="lanes">
  <div
    className="lane">
    <div
      className="lane-header"
      onClick={[Function]}>
      <div
        className="lane-add-note">
        <button
          onClick={[Function]}>
          +
        </button>
      </div>
      <input
        autoFocus={true}
        className="edit lane-name"
        defaultValue="testing"
        onBlur={[Function]}
        onEdit={[Function]}
        onKeyPress={[Function]}
        type="text" />
      <div
        className="lane-delete">
        <button
          onClick={[Function]}>
          x
        </button>
      </div>
    </div>
    <ul
      className="notes" />
  </div>
</div>
`;
